src/frontend/src/repositories/DeviceRepository.js

```diff
@@ -11,3 +11,6 @@ export default {
     return Client.get(`${resource}`)
   },
 }
+export function getDeviceDetailRoute(id) {
+  return `/devices/${id}`
+}
```

Comment:
To be more consist with the rest of the code, could you add this above as `detail(...)`